### PR TITLE
extract `:common:problem` module

### PR DESCRIPTION
**Pull request by dsvoronin**
fix incident trace

### CODE REVIEW

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportProcessorImpl.kt [hunk 1]
```diff
@@ -153,7 +156,8 @@ internal class ReportProcessorImpl(
                         incident = Incident(
                             type = Incident.Type.INFRASTRUCTURE_ERROR,
                             timestamp = now,
-                            trace = problem.asRuntimeException().stackTraceToList(),
+                            trace = problem.throwable?.stackTraceToList()
+                                ?: problem.asRuntimeException().stackTraceToList(),
```
Comment (dsvoronin) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportProcessorImpl.kt:160]: @sboishtyan that would act as before, i introduced another layer of exception on top of original cause, which is kinda useless;

File: subprojects/common/problem/src/test/kotlin/com/avito/android/ProblemKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.avito.android
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.Test
+
+internal class ProblemKtTest {
```
Comment (eugene-krivobokov) [inline subprojects/common/problem/src/test/kotlin/com/avito/android/ProblemKtTest.kt]:
```suggestion
internal class ProblemTest {
```

File: subprojects/common/problem/src/main/kotlin/com/avito/android/Problem.kt [hunk 3]
```diff
@@ -12,45 +12,45 @@ internal data class Problem(
...
-    class Builder(private val shortDescription: String, private val context: String) {
+    public class Builder(private val shortDescription: String, private val context: String) {
 
         private var because: String? = null
         private val possibleSolutions = mutableListOf<String>()
         private var documentedAt: String? = null
         private var throwable: Throwable? = null
 
-        constructor(source: Problem) : this(source.shortDescription, source.context) {
+        internal constructor(source: Problem) : this(source.shortDescription, source.context) {
```
Comment (eugene-krivobokov) [inline subprojects/common/problem/src/main/kotlin/com/avito/android/Problem.kt:26]:
```suggestion
        private constructor(source: Problem) : this(source.shortDescription, source.context) {
```
?

Comment (dsvoronin) [reply subprojects/common/problem/src/main/kotlin/com/avito/android/Problem.kt:26]: can't ; private constructor is not available for `newBuilder()`

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**